### ISSUE TITLE
after I add `display :false`, the app stoped with many exceptions.

### ISSUE THREAD

**Issue by oxf1** [grails/grails-core#9030]
![Image of error](https://cloud.githubusercontent.com/assets/976008/7900751/fad2dd08-079a-11e5-8513-2502d12dc1be.png)

my demo code is : https://github.com/xuchaoo/smonitor , you can reproduce my error by following the step:

- gradlew run
- open http://localhost:8080/stock/create, DON'T FORGET, IMPORTANT STEP!!
- modify code : remove or add `display: false` in the `static constraints` block of field `Notice`
- you can see many exceptions in the console, and the app stoped

then the app stoped with so many exceptions!!

```java
C:\code\play\smonitor>grails run-app
:compileJava UP-TO-DATE
:compileGroovy
:processResources
:classes
:findMainClass
:bootRun
Grails application running at http://localhost:8080
ERROR grails.boot.GrailsApp - Application startup failed
java.lang.IllegalStateException: Failed to introspect annotations: public static java.util.List mystock.Stock.findAll(my
stock.Stock,java.util.Map)
        at org.springframework.core.annotation.AnnotatedElementUtils.process(AnnotatedElementUtils.java:166) ~[spring-co
re-4.1.6.RELEASE.jar:4.1.6.RELEASE]
        at org.springframework.core.annotation.AnnotatedElementUtils.isAnnotated(AnnotatedElementUtils.java:73) ~[spring
-core-4.1.6.RELEASE.jar:4.1.6.RELEASE]
Exception in thread "Thread-10"         at org.springframework.core.type.StandardAnnotationMetadata.hasAnnotatedMethods(
StandardAnnotationMetadata.java:131) ~[spring-core-4.1.6.RELEASE.jar:4.1.6.RELEASE]
        at org.springframework.context.annotation.ConfigurationClassUtils.isLiteConfigurationCandidate(ConfigurationClas
sUtils.java:157) ~[spring-context-4.1.6.RELEASE.jar:4.1.6.RELEASE]
java.lang.IllegalStateException: Failed to introspect annotations: public static java.util.List mystock.Stock.findAll(my
stock.Stock,java.util.Map)
        at org.springframework.core.annotation.AnnotatedElementUtils.process(AnnotatedElementUtils.java:166)
        at org.springframework.context.annotation.ConfigurationClassUtils.checkConfigurationClassCandidate(Configuration
ClassUtils.java:108) ~[spring-context-4.1.6.RELEASE.jar:4.1.6.RELEASE]
        at org.springframework.context.annotation.ConfigurationClassPostProcessor.processConfigBeanDefinitions(Configura
tionClassPostProcessor.java:278) ~[spring-context-4.1.6.RELEASE.jar:4.1.6.RELEASE]
        at org.springframework.core.annotation.AnnotatedElementUtils.isAnnotated(AnnotatedElementUtils.java:73)
        at org.springframework.context.annotation.ConfigurationClassPostProcessor.postProcessBeanDefinitionRegistry(Conf
igurationClassPostProcessor.java:239) ~[spring-context-4.1.6.RELEASE.jar:4.1.6.RELEASE]
        at org.springframework.core.type.StandardAnnotationMetadata.hasAnnotatedMethods(StandardAnnotationMetadata.java:
131)
        at org.springframework.context.support.PostProcessorRegistrationDelegate.invokeBeanFactoryPostProcessors(PostPro
cessorRegistrationDelegate.java:119) ~[spring-context-4.1.6.RELEASE.jar:4.1.6.RELEASE]
        at org.springframework.context.annotation.ConfigurationClassUtils.isLiteConfigurationCandidate(ConfigurationClas
sUtils.java:157)
        at org.springframework.context.annotation.ConfigurationClassUtils.checkConfigurationClassCandidate(Configuration
ClassUtils.java:108)
        at org.springframework.context.support.AbstractApplicationContext.invokeBeanFactoryPostProcessors(AbstractApplic
ationContext.java:606) ~[spring-context-4.1.6.RELEASE.jar:4.1.6.RELEASE]
        at org.springframework.context.annotation.ConfigurationClassPostProcessor.processConfigBeanDefinitions(Configura
tionClassPostProcessor.java:278)
        at org.springframework.context.support.AbstractApplicationContext.refresh(AbstractApplicationContext.java:462) ~
[spring-context-4.1.6.RELEASE.jar:4.1.6.RELEASE]
        at org.springframework.context.annotation.ConfigurationClassPostProcessor.postProcessBeanDefinitionRegistry(Conf
igurationClassPostProcessor.java:239)
        at org.springframework.context.support.PostProcessorRegistrationDelegate.invokeBeanFactoryPostProcessors(PostPro
cessorRegistrationDelegate.java:119)
        at org.springframework.context.support.AbstractApplicationContext.invokeBeanFactoryPostProcessors(AbstractApplic
ationContext.java:606)
        at org.springframework.boot.context.embedded.Embe       at org.springframework.context.support.AbstractApplicati
onContext.refresh(AbstractApplicationContext.java:462)
ApplicationContext.refresh(EmbeddedWebApplicationContext.java:118) ~[spring-boot-1.2.3.RELEASE.jar:1.2.3.RELEASE]
        at org.springframework.boot.context.embedded.EmbeddedWebApplicationContext.refresh(EmbeddedWebApplicationContext
.java:118)
        at org.springframework.boot.SpringApplication.refresh(SpringApplication.java:686) ~[spring-boot-1.2.3.RELEASE.ja
r:1.2.3.RELEASE]
        at org.springframework.boot.SpringApplication.refresh(SpringApplication.java:686)
        at org.springframework.boot.SpringApplication.run(SpringApplication.java:320) ~[spring-boot-1.2.3.RELEASE.jar:1.
2.3.RELEASE]
        at org.springframework.boot.SpringApplication.run(SpringApplication.java:320)
        at grails.boot.GrailsApp.super$2$run(GrailsApp.groovy) [grails-core-3.0.1.jar:3.0.1]
        at grails.boot.GrailsApp.super$2$run(GrailsApp.groovy)
        at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method) ~[na:1.7.0_67]
        at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
        at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:57) ~[na:1.7.0_67]
        at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:57)
        at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43) ~[na:1.7.0_67]
        at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
        at java.lang.reflect.Method.invoke(Method.java:606) ~[na:1.7.0_67]
        at java.lang.   at org.springsource.loaded.ri.ReflectiveInterceptor.jlrMethodInvoke(ReflectiveInterceptor.java:1
270) [springloaded-1.2.3.RELEASE.jar:1.2.3.RELEASE]
.Method.invoke(Method.java:606)
        at org.codehaus.groovy.reflection.CachedMethod.invoke(CachedMethod.java:90) [groovy-2.4.3.jar:2.4.3]
        at org.springsource.loaded.ri.ReflectiveInterceptor.jlrMethodInvoke(ReflectiveInterceptor.java:1270)
        at groovy.lang.MetaMethod.doMethodInvoke(MetaMethod.java:324) [groovy-2.4.3.jar:2.4.3]
        at org.codehaus.groovy.reflection.CachedMethod.invoke(CachedMethod.java:90)
        at groovy.lang.MetaClassImpl.invokeMethod(MetaClassImpl.java:1207) [groovy-2.4.3.jar:2.4.3]
        at groovy.lang.MetaMethod.doMethodInvoke(MetaMethod.java:324)
        at org.codehaus.groovy.runtime.ScriptBytecodeAdapter.invokeMethodOnSuperN(ScriptBytecodeAdapter.java:130) [groov
y-2.4.3.jar:2.4.3]
        at groovy.lang.MetaClassImpl.invokeMethod(MetaClassImpl.java:1207)
        at grails.boot.GrailsApp$_enableDevelopmentModeWatch_closure1.doCall(GrailsApp.groovy:180) [grails-core-3.0.1.ja
r:3.0.1]
        at org.codehaus.groovy.runtime.ScriptBytecodeAdapter.invokeMethodOnSuperN(ScriptBytecodeAdapter.java:130)
        at grails.boot.GrailsApp$_enableDevelopmentModeWatch_closure1.doCall(GrailsApp.groovy) [grails-core-3.0.1.jar:3.
0.1]
        at grails.boot.GrailsApp$_enableDevelopmentModeWatch_closure1.doCall(GrailsApp.groovy:180)
        at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method) ~[na:1.7.0_67]
        at grails.boot.GrailsApp$_enableDevelopmentModeWatch_closure1.doCall(GrailsApp.groovy)
        at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:57) ~[na:1.7.0_67]
        at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
        at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43) ~[na:1.7.0_67]
        at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:57)
        at java.lang.reflect.Method.invoke(Method.java:606) ~[na:1.7.0_67]
        at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
        at org.springsource.loaded.ri.ReflectiveInterceptor.jlrMethodIn at java.lang.reflect.Method.invoke(Method.java:6
06)
flectiveInterceptor.java:1270) [springloaded-1.2.3.RELEASE.jar:1.2.3.RELEASE]
        at org.springsource.loaded.ri.ReflectiveInterceptor.jlrMethodInvoke(ReflectiveInterceptor.java:1270)
        at org.codehaus.groovy.reflection.CachedMethod.invoke(CachedMethod.java:90) [groovy-2.4.3.jar:2.4.3]
        at org.codehaus.groovy.reflection.CachedMethod.invoke(CachedMethod.java:90)
        at groovy.lang.MetaMethod.doMethodInvoke(MetaMethod.java:324) [groovy-2.4.3.jar:2.4.3]
        at groovy.lang.MetaMethod.doMethodInvoke(MetaMethod.java:324)
        at org.codehaus.groovy.runtime.metaclass.ClosureMetaClass.invokeMethod(ClosureMetaClass.java:292) [groovy-2.4.3.
jar:2.4.3]
        at org.codehaus.groovy.runtime.metaclass.ClosureMetaClass.invokeMethod(ClosureMetaClass.java:292)
        at groovy.lang.MetaClassImpl.invokeMethod(MetaClassImpl.java:1016) [groovy-2.4.3.jar:2.4.3]
        at groovy.lang.MetaClassImpl.invokeMethod(MetaClassImpl.java:1016)
        at groovy.lang.Closure.call(Closure.java:423) [groovy-2.4.3.jar:2.4.3]
        at groovy.lang.Closure.call(Closure.java:423)
        at groovy.lang.Closure.call(Closure.java:417) [groovy-2.4.3.jar:2.4.3]
        at groovy.lang.Closure.call(Closure.java:417)
        at groovy.lang.Closure.run(Closure.java:504) [groovy-2.4.3.jar:2.4.3]
        at groovy.lang.Closure.run(Closure.java:504)
        at java.lang.Thread.run(Thread.java:745) [na:1.7.0_67]
        at java.lang.Thread.run(Thread.java:745)
Caused by: java.lang.NullPointerException: null
Caused by: java.lang.NullPointerException
        at org.springsource.loaded.ri.ReflectiveInterceptor.jlrMethodGetDeclaredAnnotations(ReflectiveInterceptor.java:9
46) [springloaded-1.2.3.RELEASE.jar:1.2.3.RELEASE]
        at org.springsource.loaded.ri.ReflectiveInterceptor.jlrMethodGetDeclaredAnnotations(ReflectiveInterceptor.java:9
46)
        at org.springsource.loaded.ri.ReflectiveInterceptor.jlrMethodGetAnnotations(ReflectiveInterceptor.java:1502)
        at org.springsource.loaded.ri.ReflectiveInterceptor.jlrMethodGetAnnotations(ReflectiveInterceptor.java:1502) [sp
ringloaded-1.2.3.RELEASE.jar:1.2.3.RELEASE]
        at org.springsource.loaded.ri.ReflectiveInterceptor.jlrAccessibleObjectGetAnnotations(ReflectiveInterceptor.java
:1407)
        at org.springsource.loaded.ri.ReflectiveInterceptor.jlrAccessibleObjectGetAnnotations(ReflectiveInterceptor.java
:1407) [springloaded-1.2.3.RELEASE.jar:1.2.3.RELEASE]
        at org.springsource.loaded.ri.ReflectiveInterceptor.jlrAnnotatedElementGetAnnotations(ReflectiveInterceptor.java
:1331)
        at org.springsource.loaded.ri.ReflectiveInterceptor.jlrAnnotatedElementGetAnnotations(ReflectiveInterceptor.java
:1331) [springloaded-1.2.3.RELEASE.jar:1.2.3.RELEASE]
        at org.springframework.core.annotation.AnnotatedElementUtils.doProcess(AnnotatedElementUtils.java:192) ~[spring-
core-4.1.6.RELEASE.jar:4.1.6.RELEASE]
org.springframework.core.annotation.AnnotatedElementUtils.doProcess(AnnotatedElementUtils.java:192)
        at org.springframework.core.annotation.AnnotatedElementUtils.process(AnnotatedElementUtils.java:162)
        at org.springframework.core.annotation.AnnotatedElementUtils.process(AnnotatedElementUtils.java:162) ~[spring-co
re-4.1.6.RELEASE.jar:4.1.6.RELEASE]
        ... 37 more
        ... 37 common frames omitted

BUILD SUCCESSFUL

Total time: 1 mins 24.193 secs
C:\code\play\smonitor>
```

**Comment by xuchaoo** [comment]
you can find more information at https://github.com/grails3-plugins/fields/issues/2

**Comment by jeffbrown** [comment]
As I mentioned at https://github.com/grails3-plugins/fields/issues/2, I am having some trouble reproducing the problem.  I believe that it is happening, but it isn't yet clear how to make it happen.  It maybe that there is something in my environment that is preventing the problem for coming up or something in your environment that is causing it to come up.  I would like to get to the bottom of it.

I have recorded another video demonstrating what this looks like at my end.  @xuchaoo if you could look at http://jeffathome.s3.amazonaws.com/smonitor_problem.mp4 and let me know how that relates to what you are seeing, we would really appreciate your feedback.  Thanks.

**Comment by xuchaoo** [comment]
@jeffbrown  I think it's not a problem of grails-fileds.  you can try again under linux or windows. my enviroment is 
- `windows7 64` or `centos 3.13.0-32-generic #57-Ubuntu SMP Tue Jul 15 03:51:08 UTC 2014 x86_64 x86_64 x86_64 GNU/Linux`
- jdk 1.8 .0_31-b13
- grails 3.02 / grails 3.01
- and I also add groovy 2.4.3 in my PATH

another, I also found you needn't to visit http://localhost:8080/stock/create, the problem will reproduce. So I believe It's a problem of compile and reload system of grails.

**Comment by jeffbrown** [comment]
We will run some tests on Linux and Windows and see if we can get anywhere with it.

Thanks again.

**Comment by jeffbrown** [comment]
> and I also add groovy 2.4.3 in my PATH

I don't think any version of Groovy that you put on your PATH will affect the behavior of your Grails apps.

**Comment by graemerocher** [comment]
@oxf1 The repository linked to at  https://github.com/xuchaoo/smonitor doesn't exist

**Comment by oxf1** [comment]
@graemerocher  please visit https://github.com/jeffbrown/smonitor

**Comment by graemerocher** [comment]
There have been many fixes in this area in Grails 3.0.x master and the latest hibernate plugin. I tested with the latest code and experienced no exception on either OS X or Ubuntu Linux.
